Source/WebCore/svg/SVGPathByteStream.h

```diff
@@ -39,71 +40,158 @@ namespace WebCore {
+        }
+
+        friend bool operator==(const Data& a, const Data& b)
+        {
+            return a.m_bytes == b.m_bytes;
+        }
+
+        unsigned size() const { return m_bytes.size(); }
+        bool isEmpty() const { return m_bytes.isEmpty(); }
+
+    private:
+        Data(Bytes&& bytes)
+            : m_bytes(WTFMove(bytes))
+        {
+        }
+
+        Data() = default;
+        Data(const Data& data)
+            : m_bytes(data.m_bytes)
+            , m_path(data.m_path)
+        {
+        }
+
+        Bytes m_bytes;
+        mutable Path m_path;
+    };
+
+    using DataIterator = Data::Bytes::const_iterator;
+
+    SVGPathByteStream()
+        : m_data(Data::empty())
+    {
+    }
 
     SVGPathByteStream(StringView string)
+        : SVGPathByteStream()
     {
         buildSVGPathByteStreamFromString(string, *this, UnalteredParsing);
     }
 
     SVGPathByteStream(const SVGPathByteStream& other)
-        : CanMakeSingleThreadWeakPtr<SVGPathByteStream>()
+        : SVGPathByteStream()
     {
         *this = other;
     }
 
     SVGPathByteStream(SVGPathByteStream&& other)
+        : SVGPathByteStream()
     {
         *this = WTFMove(other);
     }
 
-    SVGPathByteStream(Data&& data)
+    SVGPathByteStream(Ref<Data>&& data)
         : m_data(WTFMove(data))
     {
     }
 
-    SVGPathByteStream& operator=(const SVGPathByteStream& other)
+    SVGPathByteStream(Data::Bytes&& data)
+        : m_data(Data::create(WTFMove(data)))
     {
-        if (*this == other)
-            return *this;
-        m_data = other.m_data;
-        return *this;
     }
 
-    SVGPathByteStream& operator=(SVGPathByteStream&& other)
+    SVGPathByteStream& operator=(const SVGPathByteStream& other) = default;
+    SVGPathByteStream& operator=(SVGPathByteStream&& other) = default;
+
+    friend bool operator==(const SVGPathByteStream& a, const SVGPathByteStream& b)
     {
-        if (*this == other)
-            return *this;
-        m_data = WTFMove(other.m_data);
-        return *this;
+        return a.m_data == b.m_data;
     }
 
-    friend bool operator==(const SVGPathByteStream& a, const SVGPathByteStream& b) { return a.m_data == b.m_data; }
-
     std::unique_ptr<SVGPathByteStream> copy() const
     {
         return makeUnique<SVGPathByteStream>(*this);
     }
 
-    DataIterator begin() const { return m_data.begin(); }
-    DataIterator end() const { return m_data.end(); }
+    DataIterator begin() const { return m_data->bytes().begin(); }
+    DataIterator end() const { return m_data->bytes().end(); }
+
+    void append(uint8_t byte) { m_data.access().append(byte); }
+    void append(std::span<const uint8_t> bytes) { m_data.access().append(bytes); }
+    void append(const SVGPathByteStream& other) { m_data.access().append(other.m_data->bytes()); }
+    void clear() { m_data = Data::empty(); }
+    bool isEmpty() const { return m_data->isEmpty(); }
+    unsigned size() const { return m_data->size(); }
+    void shrinkToFit() { }
```

Comment:
It's a bit surprising to have this being a no-op (even if it's not a correctness risk). We should implement it or add a comment at least.

Reply:
Changed.